src/charm.py

```diff
@@ -88,13 +89,19 @@ def _configure(self, event):
         be made
         """
         self.unit.status = MaintenanceStatus("Configuring pod")
-        plan = self._container.get_plan()
-        layer = self._pebble_layer
-        if plan.services != layer.services:
-            self._container.add_layer(CHARM_NAME, layer, combine=True)
-            self._container.restart(CHARM_NAME)
-            logger.info(f"Restarted container {CHARM_NAME}")
-        self.unit.status = ActiveStatus()
+        try:
```

Comment:
As in the (charmed-magma PR #21)[https://github.com/canonical/charmed-magma/pull/21/files] - aren't we interested in deferring the event if container is unreachable?

Reply:
Agreed, fixed